pkg/certificate/service.go

```diff
@@ -233,9 +235,11 @@ func New(ctx context.Context, idConfig *config.IdentityConfig) (daemon.Daemon, e
 					keyPEM = k8sSecretBackupKeyPEM
 					log.Infof("Successfully loaded x509 certificate from kubernetes secret")
 
-					err = idConfig.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM)
-					if err != nil {
-						log.Errorf("Failed to reload x509 certificate from identity provider: %s", err.Error())
+					if identity != nil && len(keyPEM) != 0 {
+						err = idConfig.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM)
```

Comment:
```suggestion
						errUpdate := idConfig.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM)
```

Reply:
there is a chance that this might actually overwrite err and cause problem. so its prolly better write in a different err instead.